src/shared/parse-helpers.c

```diff
@@ -196,3 +198,23 @@ int parse_socket_bind_item(
         *port_min = mn;
         return 0;
 }
+
+int parse_open_file_fields(const char *v, char **path, char **fdname, int *flags) {
+    _cleanup_strv_free_ char **parts = NULL;
+    int r;
+
+    if (!v)
+        return -EINVAL;
+
+    r = strv_split_full(&parts, v, ":", EXTRACT_DONT_COALESCE_SEPARATORS);
+    if (r < 0)
+            return r;
+    if (strv_length(parts) != 3)
+            return -EINVAL;
```

Comment:
If the user doesn't specify `fd-name` or `options`, we should just use default values for those. So we shouldn't require all 3 parts to be specified all times, which means this check needs to be adjusted.